### PR TITLE
Depend on a new Xapian version

### PR DESCRIPTION
**Pull request by ebassi**
Since we're depending on a new version of Xapian that supercedes the Xapian 1.3 copy we use currently, let's take a break from the existing API, and bump up to API series 2.0.

https://phabricator.endlessm.com/T21284

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -98,19 +103,11 @@ if get_option('default_library') != 'static'
...
 
 # Linker flags
-if host_machine.system() == 'linux'
-  foreach ldflag: [ '-Wl,-Bsymbolic', '-Wl,-z,relro', '-Wl,-z,now', ]
-    if cpp.has_argument(ldflag)
-      common_ldflags += ldflag
-    endif
-  endforeach
+if host_machine.system() == 'linux' and cpp.get_id() == 'gcc'
+  common_ldflags += [ '-Wl,-Bsymbolic', '-Wl,-z,relro', '-Wl,-z,now', ]
```
Comment (ptomato) [inline meson.build]: Would you not want these flags with Clang?

File: xapian-glib/xapian-database.cc [hunk 2]
```diff
@@ -692,18 +692,25 @@ xapian_database_add_database (XapianDatabase *db,
 /**
  * xapian_database_compact_to_path:
  * @self: A #XapianDatabase
- * @path: The path to write the compacted database to.
- * @flags: Flags to compact the database with.
+ * @path: The path to write the compacted database to
+ * @flags: Flags to compact the database with
+ * @error: return location to a #GError
```
Comment (ptomato) [inline xapian-glib/xapian-database.cc]: nitpick: return location *for* a GError

File: xapian-glib/xapian-database.cc [hunk 3]
```diff
@@ -719,30 +726,42 @@ xapian_database_compact_to_path (XapianDatabase             *self,
...
 }
 
 /**
  * xapian_database_compact_to_fd:
  * @self: A #XapianDatabase
- * @fd: The fd to write the compacted database.
- * @flags: Flags to compact the database with.
+ * @fd: The fd to write the compacted database
+ * @flags: Flags to compact the database with
+ * @error: return location to a #GError
```
Comment (ptomato) [inline xapian-glib/xapian-database.cc]: ditto

File: doc/meson.build [hunk 4]
```diff
@@ -31,25 +31,27 @@ glib_docpath = join_paths(glib_prefix, 'share', 'gtk-doc', 'html')
 
 docpath = join_paths(xapian_glib_datadir, 'gtk-doc', 'html')
 
-gnome.gtkdoc('xapian-glib',
```
Comment (ptomato) [inline doc/meson.build]: Nothing changed here except the formatting, is that intended?

File: xapian-glib/xapian-enums.h [hunk 5]
```diff
@@ -329,6 +329,7 @@ GType xapian_database_flags_get_type (void);
  * @XAPIAN_DATABASE_BACKEND_CHERT: Use the Chert backend
  * @XAPIAN_DATABASE_BACKEND_STUB: Open a stub database
  * @XAPIAN_DATABASE_BACKEND_INMEMORY: Use the "in memory" backend
+ * @XAPIAN_DATABASE_BACKEND_HONEY: Use the Honey database
```
Comment (ptomato) [inline xapian-glib/xapian-enums.h]: s/database/backend/

File: meson.build (hunk 1 above)
Comment (ebassi) [reply meson.build]: We cannot test for the presence of these flags, unlike with compiler ones; in theory, this syntax is supported by Clang, but there's really no guarantee that any non-GCC compiler will effectively support it without errors.

File: doc/meson.build (hunk 4 above)
Comment (ebassi) [reply doc/meson.build]: Yes; I kept the coding style consistent.

**Comment by ebassi** [comment]
> To be honest though I'm surprised that the CI build succeeded. I would have expected it to fail since we don't have Xapian 1.5 in the Flatpak SDK

The CI failed because the debian packaging still references an unknown option (`--with-xapian-config`). No idea why this isn't reflected in the results.

File: doc/meson.build (hunk 4 above)
Comment (ebassi) [reply doc/meson.build]: Now that you mention it, no: it wasn't intended. Just fixedit.

File: meson.build (hunk 1 above)
Comment (ptomato) [reply meson.build]: Okay then. I guess doing it properly would depend on a `meson.get_linker()` API.

**Pull request merged**